File: atcoder/javascriptENV/library/primeFactorization.js
const primeFactorization = (n) => {
  const primes = [2,3,5,7,11,13,17,19,23,29,31,37,41,43,47,53,59,61,67,71,73,79,83,89,97,101]
  const ans = []

  const isPrime = (v) => {
    if (v < 2) return false
    for (let i=2; i<=Math.floor(Math.sqrt(v)); i++)
      if (v % i === 0) return false
    return true
  }

  outer:
  while (n > 1) {
    for (let i=0; i<primes.length; i++) {
      if (n % primes[i] === 0) {
        n = n/primes[i]
        ans.push(primes[i])
        continue outer
      }
    }

    for (let i=2003; i<=n; i++) {
      if (isPrime(i) && n % i === 0) {
        n = n/i
        ans.push(i)
        break
      }
    }
  }

  return Array.from(new Set(ans))
}

File: atcoder/pythonENV/libraries/prime_factorization.py
def pf(n):
    ans = []

    while n % 2 == 0:
        ans.append(2)
        n = n / 2

    for i in range(3, math.floor(math.sqrt(n)) + 1, 2):
        while n % i == 0:
            ans.append(i)
            n = n / i

    if n > 2:
        ans.append(n)

    return list(map(int, ans))
